### ISSUE TITLE
How to accumulate gradients, and only update mask r-cnn weights after k iterations?

### ISSUE THREAD

**Issue by simaoh** [facebookresearch/detectron2#1147]
## How to accumulate gradients, and only update mask r-cnn weights after k iterations?  

I would like to change the detectron 2 codebase to only take an optimizer.step() after k iterations, accumulating the gradients over the previous k steps? (In particular k=1 would give the current behaviour)

The `run_step()` function of the `class SimpleTrainer` of `engine/train_loop.py` seems to indicate what functions need to be changed but the explanations therein (see below) are not clear to me:
https://github.com/facebookresearch/detectron2/blob/ffff8acc35ea88ad1cb1806ab0f00b4c1c5dbfd9/detectron2/engine/train_loop.py#L223-L234

Can you provide an example of how we can "wrap" the optimizer.zero_grad() and self.optimizer.step() to only happen every k iterations?

**Comment by ppwwyyxx** [comment]
> Can you provide an example of how we can "wrap" the optimizer.zero_grad() and self.optimizer.step() to only happen every k iterations?

It's a pytorch question and you can find examples of accumulate graidents on pytorch forum.
